v2/tests/log_sink_etw_2_ut/log_sink_etw_2_ut.c

```diff
@@ -0,0 +1,802 @@
+
+typedef struct log_context_property_if_get_type_CALL_TAG
+{
+    bool override_result;
+    LOG_CONTEXT_PROPERTY_TYPE call_result;
+} log_context_property_if_get_type_CALL;
+
+typedef struct MOCK_CALL_TAG
+{
+    MOCK_CALL_TYPE mock_call_type;
+    union
+    {
+        printf_CALL printf_call;
+        InterlockedCompareExchange_CALL InterlockedCompareExchange_call;
+        InterlockedExchange_CALL InterlockedExchange_call;
+        log_context_get_property_value_pair_count_CALL log_context_get_property_value_pair_count_call;
+        log_context_get_property_value_pairs_CALL log_context_get_property_value_pairs_call;
+        TraceLoggingRegister_EventRegister_EventSetInformation_CALL TraceLoggingRegister_EventRegister_EventSetInformation_call;
+        _get_pgmptr_CALL _get_pgmptr_call;
+        _tlgWriteTransfer_EventWriteTransfer_CALL _tlgWriteTransfer_EventWriteTransfer_call;
+        log_context_property_if_get_type_CALL log_context_property_if_get_type_call;
+    } u;
+} MOCK_CALL;
+
+static char* stringify_bytes(const unsigned char* bytes, size_t byte_count)
+{
+    size_t i;
+    size_t pos = 0;
+    size_t stringified_length = 3 + (byte_count * 4);
+    char* result;
+
+    if (byte_count > 0)
+    {
+        stringified_length += (byte_count - 1);
+    }
+
+    result = malloc(stringified_length);
+    if (result != NULL)
+    {
+        result[pos++] = '[';
+
+        for (i = 0; i < byte_count; i++)
+        {
+            int sprintf_result = sprintf(&result[pos], "0x%02X%s", bytes[i], (i < byte_count - 1) ? "," : "");
+            pos += sprintf_result;
+        }
+        result[pos++] = ']';
+        result[pos] = '\0';
+    }
+
+    return result;
+}
+
+static MOCK_CALL expected_calls[MAX_MOCK_CALL_COUNT];
+static size_t expected_call_count;
+static size_t actual_call_count;
+static bool actual_and_expected_match;
+
+static void setup_mocks(void)
+{
+    expected_call_count = 0;
+    actual_call_count = 0;
+    actual_and_expected_match = true;
+}
+
+int mock_printf(const char* format, ...)
+{
+    int result;
+
+    if ((actual_call_count == expected_call_count) ||
+        (expected_calls[actual_call_count].mock_call_type != MOCK_CALL_TYPE_printf))
+    {
+        actual_and_expected_match = false;
+        result = -1;
+    }
+    else
+    {
+        if (expected_calls[actual_call_count].u.printf_call.override_result)
+        {
+            result = expected_calls[actual_call_count].u.printf_call.call_result;
+        }
+        else
+        {
+            va_list args;
+            va_start(args, format);
+            // also capture the result in a variable for comparisons in tests
+            (void)vsnprintf(expected_calls[actual_call_count].u.printf_call.captured_output, sizeof(expected_calls[actual_call_count].u.printf_call.captured_output),
+                format, args);
+            va_end(args);
+
+            va_start(args, format);
+            result = vprintf(format, args);
```

Comment:
>vprintf [](http://example.com/codeflow?start=21&length=7)

Is thsi debug leftover?

Reply:
Actually doesn't look like that, it is more deep.

Reply:
If you don't want it on the screen, then you can write it in "NUL" file on windows (`/dev/nul` for Linux)`. Writes never fail there. Just saying.

Reply:
Maybe it is really intended to get on the screen and mimic the "real" calls. I got to that after seeing the next function,.

Reply:
I just wanted to have the "real" function called, yes.